coil-base/src/main/java/coil/ComponentRegistry.kt

```diff
@@ -12,71 +15,103 @@ import coil.util.MultiMutableList
  *
- * Use this class to register support for custom [Mapper]s, [MeasuredMapper]s, [Fetcher]s, and [Decoder]s.
+ * Use this class to register support for custom [Interceptor]s, [Mapper]s, [Fetcher]s, and [Decoder]s.
  */
 class ComponentRegistry private constructor(
-    internal val mappers: MultiList<Class<out Any>, Mapper<out Any, *>>,
-    internal val measuredMappers: MultiList<Class<out Any>, MeasuredMapper<out Any, *>>,
-    internal val fetchers: MultiList<Class<out Any>, Fetcher<out Any>>,
+    internal val interceptors: List<Interceptor>,
+    internal val mappers: MultiList<Mapper<out Any, *>, Class<out Any>>,
+    internal val measuredMappers: MultiList<MeasuredMapper<out Any, *>, Class<out Any>>,
+    internal val fetchers: MultiList<Fetcher<out Any>, Class<out Any>>,
     internal val decoders: List<Decoder>
 ) {
 
-    constructor() : this(emptyList(), emptyList(), emptyList(), emptyList())
+    constructor() : this(emptyList(), emptyList(), emptyList(), emptyList(), emptyList())
 
     fun newBuilder() = Builder(this)
 
     class Builder {
 
-        private val mappers: MultiMutableList<Class<out Any>, Mapper<out Any, *>>
-        private val measuredMappers: MultiMutableList<Class<out Any>, MeasuredMapper<out Any, *>>
-        private val fetchers: MultiMutableList<Class<out Any>, Fetcher<out Any>>
+        private val interceptors: MutableList<Interceptor>
+        private val mappers: MultiMutableList<Mapper<out Any, *>, Class<out Any>>
+        private val measuredMappers: MultiMutableList<MeasuredMapper<out Any, *>, Class<out Any>>
+        private val fetchers: MultiMutableList<Fetcher<out Any>, Class<out Any>>
         private val decoders: MutableList<Decoder>
 
         constructor() {
+            interceptors = mutableListOf()
             mappers = mutableListOf()
             measuredMappers = mutableListOf()
             fetchers = mutableListOf()
             decoders = mutableListOf()
         }
 
         constructor(registry: ComponentRegistry) {
+            interceptors = registry.interceptors.toMutableList()
             mappers = registry.mappers.toMutableList()
             measuredMappers = registry.measuredMappers.toMutableList()
             fetchers = registry.fetchers.toMutableList()
             decoders = registry.decoders.toMutableList()
         }
 
+        /** Register an [Interceptor]. */
+        fun add(interceptor: Interceptor) = apply {
+            interceptors += interceptor
+        }
+
         /** Register a [Mapper]. */
-        inline fun <reified T : Any> add(mapper: Mapper<T, *>) = add(T::class.java, mapper)
+        inline fun <reified T : Any> add(mapper: Mapper<T, *>) = add(mapper, T::class.java)
 
         @PublishedApi
-        internal fun <T : Any> add(type: Class<T>, mapper: Mapper<T, *>) = apply {
-            mappers += type to mapper
+        internal fun <T : Any> add(mapper: Mapper<T, *>, type: Class<T>) = apply {
```

Comment:
Changed the parameter order so the mapper/fetcher is `component0` instead of `component1`. This change is transparent for Kotlin users.